pkg/manifest/manifest_loader.go

```diff
@@ -0,0 +1,296 @@
+package manifest
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/dynatrace-oss/dynatrace-monitoring-as-code/pkg/util/files"
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v2"
+)
+
+type project struct {
+	Name string  `yaml:"name"`
+	Type *string `yaml:"type,omitempty"`
+	Path string  `yaml:"path"`
+}
+
+type tokenConfig struct {
+	Type   *string                `yaml:"type,omitempty"`
+	Config map[string]interface{} `yaml:",inline"`
+}
+
+type environment struct {
+	Name  string      `yaml:"name"`
+	Url   string      `yaml:"url"`
+	Token tokenConfig `yaml:"token"`
+}
+
+type group struct {
+	Group   *string       `yaml:"group"`
+	Entries []environment `yaml:"entries"`
+}
+
+type manifest struct {
+	Projects     []project `yaml:"projects"`
+	Environments []group   `yaml:"environments"`
+}
+
+func LoadManifest(fs afero.Fs, manifestPath string) (Manifest, []error) {
+	exists, err := files.DoesFileExist(fs, manifestPath)
+
+	if err != nil {
+		return Manifest{}, []error{err}
+	}
+
+	if !exists {
+		return Manifest{}, []error{fmt.Errorf("path `%s` is either no file or does not exists", manifestPath)}
+	}
+
+	if !isYaml(manifestPath) {
+		return Manifest{}, []error{fmt.Errorf("file `%s` is not yaml", manifestPath)}
+	}
+
+	data, err := afero.ReadFile(fs, manifestPath)
```

Comment:
If `len(data) == 0` no need to proceed here

Reply:
Something seems broken with this comment.